include/hyprutils/utils/FileDescriptor.hpp

```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+namespace Hyprutils {
+    namespace Utils {
+        class CFileDescriptor {
+          public:
+            CFileDescriptor() = default;
+            explicit CFileDescriptor(int fd);
+            CFileDescriptor(CFileDescriptor&&);
+            CFileDescriptor& operator=(CFileDescriptor&&);
+            ~CFileDescriptor();
+
+            bool operator==(const CFileDescriptor& rhs) const {
+                return m_fd == rhs.m_fd;
+            }
+
+            bool            isValid() const;
+            int             get() const;
+            int             take();
+            void            reset();
+            CFileDescriptor duplicate() const;
+
+            bool            isReadable() const;
+            bool            isClosed() const;
```

Comment:
can we add a method for adding / getting fd flags? (fcntl)

Reply:
added a getFlags and setFlags if these are how you had it in your mind.

Reply:
yes